akka-stream/src/main/scala/akka/stream/javadsl/Framing.scala

```diff
@@ -87,13 +87,46 @@ object Framing {
    * @param byteOrder The ''ByteOrder'' to be used when decoding the field
    */
   def lengthField(
-    fieldLength:        Int,
-    fieldOffset:        Int,
-    maximumFrameLength: Int,
+    fieldLength:        Integer,
+    fieldOffset:        Integer,
+    maximumFrameLength: Integer,
```

Comment:
this was a misunderstanding, those should still be `Int`, i.e. primitive `int` in java, the problem was only in `Function2[Array[Byte], Integer, Integer]`

remember to update MiMa also